File: newsmania-backend/src/main/java/com/newsmania/newsmania_backend/service/CommentService.java
package com.newsmania.newsmania_backend.service;

import com.newsmania.newsmania_backend.entity.Comment;
import com.newsmania.newsmania_backend.repository.CommentRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CommentService {

    private final CommentRepository commentRepository;

    public CommentService(CommentRepository commentRepository) {
        this.commentRepository = commentRepository;
    }

    // Add Comment
    public Comment addComment(Comment comment) {
        return commentRepository.save(comment);
    }

    // Get Comments by News ID
    public List<Comment> getComments(Long newsId) {
        return commentRepository.findByNewsId(newsId);
    }

    // Update Comment
    public Comment updateComment(Long id, Comment updatedComment) {

        Comment comment = commentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Comment not found"));

        comment.setComment(updatedComment.getComment());

        return commentRepository.save(comment);
    }

    // Delete Comment
    public String deleteComment(Long id) {

        commentRepository.deleteById(id);

        return "Comment deleted successfully";
    }
}

File: newsmania-frontend/src/services/commentService.js
import api from "./api";

export const getComments = async (newsId) => {
  const response = await api.get(`/comments/${newsId}`);
  return response.data;
};

export const addComment = async (commentData) => {
  const response = await api.post("/comments", commentData);
  return response.data;
};

export const updateComment = async (id, commentData) => {
  const response = await api.put(`/comments/${id}`, commentData);
  return response.data;
};

export const deleteComment = async (id) => {
  const response = await api.delete(`/comments/${id}`);
  return response.data;
};